src/utils/form-data-stream.js

```diff
@@ -0,0 +1,61 @@
+import {Readable} from 'stream';
+
+import {isBlob} from './is.js';
+
+class FormDataStream extends Readable {
+	constructor(form, boundary) {
+		super();
+
+		this._carriage = '\r\n';
+		this._dashes = '-'.repeat(2);
+		this._boundary = boundary;
+		this._form = form;
+		this._curr = this._readField();
+	}
+
+	_getHeader(name, field) {
+		let header = '';
+
+		header += `${this._dashes}${this._boundary}${this._carriage}`;
+		header += `Content-Disposition: form-data; name="${name}"`;
+
+		if (isBlob(field)) {
+			header += `; filename="${field.name}"${this._carriage}`;
+			header += `Content-Type: ${field.type || 'application/octet-stream'}`;
+		}
+
+		return `${header}${this._carriage.repeat(2)}`;
+	}
+
+	async * _readField() {
+		for (const [name, field] of this._form) {
+			yield this._getHeader(name, field);
+
+			if (isBlob(field)) {
+				yield * field.stream();
+			} else {
+				yield field;
+			}
+
+			yield this._carriage;
+		}
+
+		yield `${this._dashes}${this._boundary}${this._dashes}${this._carriage.repeat(2)}`;
+	}
+
+	_read() {
+		const onFulfilled = ({done, value}) => {
+			if (done) {
+				return this.push(null);
+			}
+
+			this.push(Buffer.isBuffer(value) ? value : Buffer.from(String(value)));
+		};
+
+		const onRejected = err => this.emit('error', err);
+
+		this._curr.next().then(onFulfilled).catch(onRejected);
+	}
+}
+
+export default FormDataStream;
```

Comment:
Yes, just bump `engines` a little more to Node 10.17 👍

Reply:
I will try this out. Perhaps I also will adopt this for the formdata-node package 👍

Reply:
fyi, we have been thinking about allowing iterator in the core before and doing "iterator to stream" ourself to allow for both whatwg and node streams (to try and solve https://github.com/node-fetch/node-fetch/issues/647, at least for the Request). (since both are asyncIterable) and yields uint8arrays (i [proposed](https://github.com/whatwg/fetch/issues/809) iterator as body to the spec and it later got closed by an ["iterator to stream"](https://github.com/whatwg/streams/issues/1018) solution instead)

so the form-data encoding could potential just be a iterator as it is (and converted later down the line)

```js
fetch(url, {
  body: asyncIterator
})
```

this would make the node-fetch more "loose" and not so strict.

Reply:
I did similar thing for my FormData implementation: it can take both node and whatwg stream as field's value.
Also, replaced FormDataStream with formDataIterator in the PR. It seem to work fine.